### PR TITLE
514 update validation interactor and sequence

### PR DESCRIPTION
**Pull request by sadlerw**
* add interactor and specs
* add action and spec
* add sequence

### CODE REVIEW

**Review by codyseibert — APPROVED** [review]

File: web-client/src/tests/validateCaseDetailAction.test.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  }),
+};
+
+presenter.providers.path = {
+  success: successStub,
+  error: errorStub,
+};
+
+describe('validateCaseDetail', async () => {
+  it('should call the path succes when no errors are found', async () => {
```
Comment (kkoskelin) [inline web-client/src/tests/validateCaseDetailAction.test.js]: `success`

File: web-client/src/tests/validateCaseDetailAction.test.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+          caseId: '123',
+        },
+      },
+      modules: {
+        presenter,
+      },
+    });
+    expect(validateCaseDetailStub.getCall(0).args[0].caseDetail).toMatchObject({
+      caseId: '123',
+      irsNoticeDate: '2009-10-13',
```
Comment (kkoskelin) [inline web-client/src/tests/validateCaseDetailAction.test.js:44]: Could we use date values in the call to `runAction` which demonstrates that the irsNoticeDate also gets zero-padded correctly?

**Review by kkoskelin — APPROVED** [review]
Fix the spelling error before it gets to W.  :)

File: web-client/src/tests/validateCaseDetailAction.test.js (hunk 2 above)
Comment (sadlerw) [reply web-client/src/tests/validateCaseDetailAction.test.js:44]: added

**Pull request merged**